source/scripts/Provider/EthProvider.ts

```diff
@@ -18,10 +18,15 @@ export const EthProvider = (host: string) => {
   const sendTransaction = async (params: ITransactionParams) => {
     const tx = params;
     const {
-      ethereumTransaction: { web3Provider },
+      ethereumTransaction: { contentScriptWeb3Provider },
     } = getController().wallet;
-    const validateTxToAddress = await validateEOAAddress(tx.to, web3Provider);
-    const isLegacyTx = !(await verifyNetworkEIP1559Compatibility(web3Provider));
+    const validateTxToAddress = await validateEOAAddress(
+      tx.to,
+      contentScriptWeb3Provider
+    );
+    const isLegacyTx = !(await verifyNetworkEIP1559Compatibility(
```

Comment:
verifyNetworkEIP1559Compatibility is a promise that might return an error in case the request fails, do we need a try-catch block around here to cover this case too?

Reply:
No, we don't because inside `verifyNetworkEIP1559Compatibility` it was implemented. So, if it break with something wrong, the trycatch will stop code flow with `throw error` structure. It works fine.

Reply:
I mean, the sendTransaction will keep the normal execution even the value is an error. And here's my point If something has an error should we break the sendTransaction or keep the next steps? I am not referring to the verify fn catch block but to the sendTransaction behavior after the error

Reply:
If any function inside sendTransaction fails, we stop the function flow, return an error, and it will not proceed to the next steps

Reply:
To clarify, the change related to the sendTransaction function was simply the replacement of the old web3Provider with the contentScriptWeb3Provider

Reply:
Yes, I know you didn't change that, I am just looking for possible errors in this old flow. I mean I know there are a few `if` blocks that will avoid keeping the flow if the error happens, but again, just wondering if we should break this fn or not, probably not a big deal at all.